sources/packages/backend/libs/sims-db/src/entities/education-program-offering.model.ts

```diff
@@ -1,295 +1,287 @@
+ */
+export const OFFERING_NAME_MAX_LENGTH = 50;
+export const OFFERING_WIL_TYPE_MAX_LENGTH = 50;
+
+/**
+ * The main resource table to store education programs offerings related information.
+ * Education programs offering is for a particular program and location.
+ */
+@Entity({ name: TableNames.EducationProgramsOfferings })
+export class EducationProgramOffering extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+  @PrimaryGeneratedColumn()
+  id: number;
+  /**
+   * Offering name
+   */
+  @Column({
+    name: "offering_name",
+  })
+  name: string;
+  /**
+   * Offering Study start date
+   */
+  @Column({
+    name: "study_start_date",
+    type: "date",
+    nullable: true,
+  })
+  studyStartDate: string;
+  /**
+   * Offering Study end date
+   */
+  @Column({
+    name: "study_end_date",
+    type: "date",
+    nullable: true,
+  })
+  studyEndDate: string;
+  /**
+   * Offering Actual Tuition Costs
+   */
+  @Column({
+    name: "actual_tuition_costs",
+    nullable: true,
+  })
+  actualTuitionCosts: number;
+  /**
+   * Offering Program Related Costs
+   */
+  @Column({
+    name: "program_related_costs",
+    nullable: true,
+  })
+  programRelatedCosts: number;
+  /**
+   * Offering Mandatory Fees
+   */
+  @Column({
+    name: "mandatory_fees",
+    nullable: true,
+  })
+  mandatoryFees: number;
+  /**
+   * Offering Exceptional Expenses
+   */
+  @Column({
+    name: "exceptional_expenses",
+    nullable: true,
+  })
+  exceptionalExpenses: number;
+  /**
+   * How Offering is Delivered like Onsite, Online, Blended
+   */
+  @Column({
+    name: "offering_delivered",
+  })
+  offeringDelivered: string;
+  /**
+   * Offering does not have Study Breaks?
+   */
+  @Column({
+    name: "lacks_study_breaks",
+  })
+  lacksStudyBreaks: boolean;
+  /**
+   * Related program.
+   */
+  @OneToOne(() => EducationProgram, { eager: false, cascade: true })
+  @JoinColumn({
+    name: "program_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  educationProgram: EducationProgram;
+  /**
+   * Related location.
+   */
+  @OneToOne(() => InstitutionLocation, { eager: false, cascade: true })
+  @JoinColumn({
+    name: "location_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  institutionLocation: InstitutionLocation;
+  /**
+   * Defines the type of the offering.
+   */
+  @Column({
+    name: "offering_type",
+    nullable: false,
+    type: "enum",
+    enum: OfferingTypes,
+    enumName: "OfferingTypes",
+  })
+  offeringType: OfferingTypes;
+  /**
+   *offering_intensity decides if offering is Full-Time or Part-Time.
+   */
+  @Column({
+    name: "offering_intensity",
+    nullable: false,
+  })
+  offeringIntensity: OfferingIntensity;
+
+  /**
+   * Offering year of study.
+   */
+  @Column({
+    name: "year_of_study",
+  })
+  yearOfStudy: number;
+
+  /**
+   * Determines if the offering has WIL component.
+   * */
```

Comment:
There is white space added between the `* *` that is not part of the main branch right now.